### PR TITLE
Mention ffi-navigator support in Python README

### PR DESCRIPTION
**Pull request by masahi**
Hi @yuanming-hu , to easily navigate between python and cpp, I've just added support for taichi in [ffi-navigator](https://github.com/tqchen/ffi-navigator) tool in [this commit](https://github.com/tqchen/ffi-navigator/pull/32) (the large diff there comes from testing code for dummy repo, the actual implementation is only [30 lines](https://github.com/tqchen/ffi-navigator/blob/master/python/ffi_navigator/dialect/taichi.py#L7-L38)).

Their gif demos on their README (emacs one recorded by me!) should give you a good sense of what ffi-navigator can do. I happen to have a detailed knowledge of the project so I was able to add taichi support with minimum effort. 

Right now it can jump from python symbols such as
* `ti.core.global_var_expr_from_snode`
* `taichi_lang_core.expr_add`, `taichi_lang_core.create_kernel`
* `tc_core.Array2DVector4`

to their definition in cpp files where they are exported via pybind. Once you are in cpp, you can use any cpp LSP client such as ccls, clangd to navigate within cpp.

### CODE REVIEW

**Comment by yuanming-hu** [comment]
Amazing! This will allow new developers to quickly master the code structure of Taichi. Thanks for offering this.

File: python/taichi/README.md [hunk 1]
```diff
@@ -267,7 +267,7 @@ Embedding the language in `python` has the following advantages:
 - Easy to learn. Python itself is very easy to learn, so is PyTaichiLang.
 - Easy to run. No ahead-of-time compilation is needed.
 - It allows people to reuse existing python infrastructure:
-   - IDEs. A python IDE simply works for TaichiLang, with syntax highlighting, checking, and autocomplete.
+   - IDEs. A python IDE simply works for TaichiLang, with syntax highlighting, checking, and autocomplete. For taichi developers, [ffi-navigator](https://github.com/tqchen/ffi-navigator) allows you to jump from C++ extension symbols exported via taichi_core in Python code to their defintion in C++.
```
Comment (yuanming-hu) [inline python/taichi/README.md]: The [Contributor Guide](https://taichi.readthedocs.io/en/latest/contributor_guide.html#tips-on-taichi-compiler-development) with source at
https://github.com/taichi-dev/taichi/blob/master/docs/contributor_guide.rst#tips-on-taichi-compiler-development seems to be a better place for this tip :-)

Comment (yuanming-hu) [reply python/taichi/README.md]: `python/taichi/README.md` is a very old introduction to Taichi. I should have removed this file. Sorry about the confusion!

**Comment by yuanming-hu** [comment]
Great, thanks!

File: python/taichi/README.md (hunk 1 above)
Comment (masahi) [reply python/taichi/README.md]: Updated.

**Pull request merged**